pkg/builder/common.go

```diff
@@ -100,7 +100,26 @@ func ApplyReplacements(build *v1alpha1.Build, replacements map[string]string) *v
 			steps[i].Env = applyEnvOverride(steps[i].Env, buildTmpl.Env)
 		}
 	}
-	return build
+
+	// Apply variable expansion to volumes.
+	if build.Spec.Volumes != nil {
+		volumeBytes, err := json.Marshal(build.Spec.Volumes)
+		if err != nil {
+			return nil, err
+		}
+
+		volumes := []corev1.Volume{}
+
+		err = json.Unmarshal([]byte(applyReplacements(string(volumeBytes))), &volumes)
```

Comment:
This isn't a very elegant way to apply these replacements, and it could conceivably result bugs with unexpected replacements.

I think it would be much better to iterate through the build's volumes and `applyReplacements` to each volume's name. Do we expect to need to apply replacements inside volume sources too?

Reply:
you are right, it will cause unexpected replacement. By just replacing volume's name makes no sense. In #296 , i think that he wants to replace something in volume sources like configMap's name.